pkg/controller/plan/context/migration.go

```diff
@@ -0,0 +1,153 @@
+package context
+
+import (
+	"context"
+	liberr "github.com/konveyor/controller/pkg/error"
+	api "github.com/konveyor/forklift-controller/pkg/apis/forklift/v1alpha1"
+	"github.com/konveyor/forklift-controller/pkg/apis/forklift/v1alpha1/snapshot"
+	"github.com/konveyor/forklift-controller/pkg/controller/provider/web"
+	core "k8s.io/api/core/v1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+//
+// Factory.
+func New(client k8sclient.Client, plan *api.Plan, migration *api.Migration) (ctx *Context, err error) {
+	ctx = &Context{
+		Client:    client,
+		Plan:      plan,
+		Migration: migration,
+	}
+	err = ctx.build()
+	if err != nil {
+		err = liberr.Wrap(err)
+		return
+	}
+
+	return
+}
+
+//
+// Plan execution context.
+type Context struct {
```

Comment:
Most of the objects involved in Plan execution (migration run), need many of the same source and destination objects.  The `Context` is an object that builds itself and can be passed around and/or aggregated into other objects such as the `Migration`, `Kubevirt` and `Builder` objects.